### PR TITLE
ENH lint for no `.ci_support` files and thus no package builds

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Added a ``news`` entry

xref: https://github.com/regro/cf-scripts/issues/2627

### CODE REVIEW

File: tests/test_lint_recipe.py
Comment (beckermr) [inline tests/test_lint_recipe.py:1779]:
```suggestion
        ci_support_dir = os.path.join(feedstock_dir, ".ci_support")
```

**Comment by beckermr** [comment]
@conda-forge/core This PR is ready for review!

File: conda_smithy/lint_recipe.py
Comment (h-vetinari) [inline conda_smithy/lint_recipe.py:1202]: Can this ever be the intended behaviour? I don't think that's possible/reasonable? Perhaps easier to just delete this line...
```suggestion
```

**Comment by beckermr** [comment]
Done! Shall we merge @h-vetinari!

File: conda_smithy/lint_recipe.py
Comment (beckermr) [inline conda_smithy/lint_recipe.py:1201]:
```suggestion
                "The feedstock has no `.ci_support` files and thus will not build any packages."
```

**Review by h-vetinari — APPROVED** [review]
👍

**Comment by jakirkham** [comment]
Thanks Matt! 🙏

This is a great idea. Even outside the case you identified, we also periodically have maintainers come to us say their feedstock didn't upload anything when they effectively skipped all builds. Hopefully this will help flag those cases before they merge such PRs

Wonder if there is a way to catch this sort of issue in `staged-recipes` too

**Pull request merged**
